### PR TITLE
fix: do not show dialog in ImportMedia screen (AR-3274)

### PR DESCRIPTION
**Pull request by borichellow**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-3274" title="AR-3274" target="_blank">BugAR-3274</a>  A "this account was used on ..." dialog is shown in the media share extension
  </td></table>
  <br />
 

# What's new in this PR?

### Issues

NewClient dialog is shown even in ImportMedia, while shouldn't.

### Causes (Optional)

was not such a requirement. 

### Solutions

Implement it, by adding `newClientDialogRemembered` into `GlobalAppState` to remember the dialog data if an event about new Client comes while user is in ImportMedia screen. 
AND
observe CurrentScreen to display that dialog once user goes to some other screen.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1833/3/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt [hunk 1]
```diff
@@ -465,6 +502,18 @@ class WireActivityViewModel @Inject constructor(
             }
         }
     }
+
+    private fun CurrentScreen.isGlobalDialogAllowed(): Boolean = when (this) {
+        CurrentScreen.ImportMedea -> false
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt]: typo

Comment (gongracr) [reply app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt]:
```suggestion
        CurrentScreen.ImportMedia -> false
```

File: app/src/main/kotlin/com/wire/android/util/CurrentScreenManager.kt [hunk 2]
```diff
@@ -115,6 +115,9 @@ sealed class CurrentScreen {
     // Incoming call screen is opened
     data class IncomingCallScreen(val id: QualifiedID) : CurrentScreen()
 
+    // Import medea screen is opened
+    object ImportMedea : CurrentScreen()
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/util/CurrentScreenManager.kt]:
```suggestion
    // Import media screen is opened
    object ImportMedia : CurrentScreen()
```

File: app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt [hunk 3]
```diff
@@ -491,4 +533,5 @@ data class GlobalAppState(
     val updateAppDialog: Boolean = false,
     val conversationJoinedDialog: JoinConversationViaCodeState? = null,
     val newClientDialog: NewClientData? = null,
+    val newClientDialogRemembered: NewClientData? = null,
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt:539]: I am still confused about what does this variable mean. Maybe some extra documentation will be helpful here.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1833/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1833/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6663950)(6663950)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6663950)(6663950)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1833/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6665283)(6665283)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6665283)(6665283)-staging-compat.apk)

File: app/src/test/kotlin/com/wire/android/ui/WireActivityViewModelTest.kt [hunk 4]
```diff
@@ -591,6 +601,48 @@ class WireActivityViewModelTest {
...
+        assertEquals(null, viewModel.globalAppState.newClientDialog)
+
+        assertEquals(
+            NewClientData.CurrentUser(TestClient.CLIENT.registrationTime?.toIsoDateTimeString()!!, TestClient.CLIENT.displayName()),
+            viewModel.globalAppState.newClientDialogRemembered
+        )
+    }
+
+    @Test
+    fun `given newClient is registered when current screen changed to ImportMedea, then remember NewClient dialog state`() {
```
Comment (alexandreferris) [inline app/src/test/kotlin/com/wire/android/ui/WireActivityViewModelTest.kt:628]:
```suggestion
    fun `given newClient is registered when current screen changed to ImportMedia, then remember NewClient dialog state`() {
```

**Review by alexandreferris — APPROVED** [review]
Nice 👌🏻 

just a comment on a typo

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1833/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6707242)(6707242)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6707242)(6707242)-staging-compat.apk)

**Review by MohamadJaara — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1833/8/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6709010)(6709010)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6709010)(6709010)-staging-compat.apk)

**Pull request merged**